application/views/ride_views.py

```diff
@@ -263,23 +262,50 @@ def put(self, rideId, requestId):
         """Driver can accept or reject the ride offer."""
         try:
             data = request.get_json()
-            act = True
+            action = False
             response = ''
+            # check whether driver already accepted the offer.
+            query = "select status from requests where req_id='{}'".format(requestId)
+            result = db.execute(query)
+            status = result.fetchone()
+
+            query = "select available_space from rides where ride_id='{}'"\
+                        .format((rideId))
+            result = db.execute(query)
+            seats = result.fetchone()
+            available_seats = seats[0]
+
+            # check for action to take
             if data['action'].lower() == 'accept':
-                act = True
-                response = {'message': 'Request accept'}
+                if status[0]:
+                    return {'message': 'You already accepted this user request'},403
+                action = True                
+                # Decrement the available seats by one
+                available_seats -= 1
+                
+                # set message to be returned to user after request update cycle
+                response = {'message': 'Request accepted'}
             else:
-                act = False
+                # Reject an already accepted request; this means available seats should be incremented
+                if status[0]:
+                    available_seats += 1
+                else:
+                    return {'message': 'Request is rejected'}
```

Comment:
If a request is rejected without having been accepted first - does the query to update the stats run? because the return statement  should halt execution

Reply:
@Bopchy 
#### This is the thought I have for this.
#### if `status[0] is True`
-> means the request had been `accepted` and now driver wants to revert it.
To take into account the new available seat, we need to increment the available space so that passengers can request to join the ride.

#### The else part
- Here, we don't need to proceed with the execution since the ride is already marked as rejected.
Am thinking of changing the message for this since it gives an illusion that the request has been executed.

#### Response to your Question you raised above.
The query to update stats does not run since we are already returning after finding out that the request is already rejected.
I hope I have answered you. If not so, let me know.